pkg/apis/core/validation/exposureclass.go

```diff
@@ -17,17 +17,40 @@ package validation
 import (
 	"github.com/gardener/gardener/pkg/apis/core"
 
+	apivalidation "k8s.io/apimachinery/pkg/api/validation"
 	"k8s.io/apimachinery/pkg/util/validation/field"
 )
 
 // ValidateExposureClass validates a ExposureClass object.
 func ValidateExposureClass(exposureClass *core.ExposureClass) field.ErrorList {
-	allErrs := field.ErrorList{}
+	var allErrs = field.ErrorList{}
+
+	if len(exposureClass.Handler) == 0 {
+		allErrs = append(allErrs, field.Invalid(field.NewPath("handler"), exposureClass.Name, "exposure class handler cannot be empty"))
+	}
+
+	return append(allErrs, validateExposureClassScheduling(exposureClass.Scheduling, field.NewPath("scheduling"))...)
+}
+
+func validateExposureClassScheduling(scheduling *core.ExposureClassScheduling, fldPath *field.Path) field.ErrorList {
+	var allErrs = field.ErrorList{}
+
+	if scheduling == nil {
+		return append(allErrs, field.Required(fldPath, "exposure class need to have a scheduling configuration"))
+	}
+
+	if (scheduling.SeedSelector == nil && len(scheduling.Tolerations) == 0) || (scheduling.SeedSelector != nil && len(scheduling.Tolerations) > 0) {
```

Comment:
Why is this constraint, i.e., why don't we allow having `ExposureClass`es with neither a seed selector/tolerations?

Reply:
I agree we should allow ExposureClasses without `scheduling`. Those ExposureClasses would select all available Seeds. Then I guess the `scheduling section should stay a pointer and be optional.

Reply:
Removed this validation as `scheduling` is now optional.